deploy/Dockerfile

```diff
@@ -1,3 +1,19 @@
+FROM golang:1.22-alpine3.19 AS builder
+
+RUN apk update && \
+    apk add --no-cache \
+    curl \
+    docker-cli \
+    bash \
+    make \
+    git \
+    g++
+
+WORKDIR /work
+COPY . .
+
+RUN make build
```

Comment:
@majst01 I'm not sure, how this change worked for you becuase you're copying the content of the /deploy folder into the container which doesn't include the source code and make file but you're trying to build binary which requires the source code and make file.  Ideally, we want to keep the binary build up outside the Dockerfile.  This is the reason for the current CI failure, i removed this change from this PR #44 and CI jobs are passing.

Reply:
The `COPY . . ` instruction copies the whole directory into the container. The Build succeeded in the github CI as you can see here in the pipeline. As i dont know how your CI works and the logs are not visible i can not tell why this  failed. 

Building binaries inside a multistage container build is best practice, because you do not relay on the CI infrastructure which is unknown for contributors and might change.